src/lib/proxyCache.js

```diff
@@ -0,0 +1,71 @@
+/*****
+ License
+ --------------
+ Copyright © 2020-2024 Mojaloop Foundation
+
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0
+ (the "License") and you may not use these files except in compliance with the [License](http://www.apache.org/licenses/LICENSE-2.0).
+
+ You may obtain a copy of the License at [http://www.apache.org/licenses/LICENSE-2.0](http://www.apache.org/licenses/LICENSE-2.0)
+
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the [License](http://www.apache.org/licenses/LICENSE-2.0).
+
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * INFITX
+ - Steven Oderayi <steven.oderayi@infitx.com>
+ --------------
+ ******/
+
+'use strict'
+
+const { createProxyCache } = require('@mojaloop/inter-scheme-proxy-cache-lib')
+const Config = require('./config.js')
+
+let proxyCache
+
+const connect = async () => {
+  return !proxyCache?.isConnected && getCache().connect()
+}
+
+const disconnect = async () => {
+  return proxyCache?.isConnected && proxyCache.disconnect()
+}
+
+const getCache = () => {
+  if (!proxyCache) {
+    proxyCache = Object.freeze(
+      createProxyCache(Config.PROXY_CACHE_CONFIG.type, Config.PROXY_CACHE_CONFIG.proxyConfig)
+    )
+  }
+  return proxyCache
+}
+
+const getConnectedCache = async () => {
+  await connect()
+  return proxyCache
+}
+
+const getClient = async () => {
+  return (await getConnectedCache()).client
+}
+
+module.exports = {
```

Comment:
Why do we need this wrapper on top of _proxyCache_?

Reply:
Because it requires initialization (configuration and connection) before it can be used.

Reply:
but we do the same without wrapper:
https://github.com/mojaloop/account-lookup-service/pull/484/files#diff-79f4c7cdc0b026f6e556196f5911a449157807cffdb3f9d33f0b5404b4ffc2f3R98

Seems, you tend to use singletone pattern (with hardcoded dependencies) rather then dependency injection